### PR TITLE
Skip docparams checker on nodes within exempted functions.

### PR DESCRIPTION
**Pull request by shvenkat**
Certain functions have a "docstring exemption". This includes functions whose name matches the `no-docstring-rgx` option, or whose docstring is not longer than the `docstring-min-length` option (if set to a positive value). These functions are partially exempt from the `parameter_documentation` checker. This PR fixes this checker to complete the exemption.

In `pylint/extensions/docparams.py`, functions matching either of the above options are skipped by the `visit_functiondef` method. Here, this logic is also applied in the other `visit_*` methods: `visit_raise`, `visit_return`, `visit_yield` and `visit_yieldfrom`.

## Steps

- [ ] Add yourself to CONTRIBUTORS if you are a new contributor.
- [ ] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [ ] Write a good description on what the PR does.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |

### CODE REVIEW

File: pylint/extensions/docparams.py [hunk 1]
```diff
@@ -202,25 +202,30 @@ class DocstringParameterChecker(BaseChecker):
...
-            return
+            return True
 
         # skip functions smaller than 'docstring-min-length'
         lines = checker_utils.get_node_last_lineno(node) - node.lineno
         max_lines = get_global_option(self, "docstring-min-length")
-        if max_lines > -1 and lines < max_lines:
+        return max_lines > -1 and lines < max_lines
+
+    def visit_functiondef(self, node):
```
Comment (Pierre-Sassoulas) [inline pylint/extensions/docparams.py:219]: We dropped python 3.5 some time ago ;)

```suggestion
    def visit_functiondef(self, node: astroid.scoped_nodes.Function) -> None:
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thank you for the merge request, this looks nice. Could you add some typing and maybe create a test ? Maybe checking the ``call_count`` on a mock?

**Comment by shvenkat** [comment]
Thanks for the typing suggestion. I accepted it as a commit here.

> Maybe checking the call_count on a mock?

I have no idea how to do that :) Any suggestions or pointers?

**Comment by Pierre-Sassoulas** [comment]
>Any suggestions or pointers?

Sure, you can read more about call_count [here](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_count)

**Comment by n-thumann** [comment]
Any update on this? I'd love so see this merged :v:

**Comment by Pierre-Sassoulas** [comment]
> Any update on this?

This need a rebase, tests and typing, it seems @shvenkat is not available to do it but I would merge it if someone build on it.

**Comment by DanielNoord** [comment]
I'm going to close this as it has become stale.

@n-thumann If you can provide a code example in which this would benefit you I would be willing to take a look at this again. For now, I don't immediately see how this would benefit us. I believe the `visit_raise` depend on us actually finding a docstring in the `visit_functiondef` function, so if we skip there because of regex we should also skip in `visit_raise`.

**Pull request closed without merging**